src/TheThingsNetwork.cpp

```diff
@@ -245,47 +220,33 @@ int TheThingsNetwork::poll(int port, bool confirm) {
 }
 
 void TheThingsNetwork::fillAirtimeInfo() {
-  getInfo(readValue(F("radio get sf")), 1);
-  getInfo(readValue(F("radio get bw")), 2);
-  getInfo(readValue(F("radio get prlen")), 3);
-  getInfo(readValue(F("radio get crc")), 4);
-  getInfo(readValue(F("radio get cr")), 5);
-  this->info.sf >= 11 ? this->info.de = 1 : this->info.de = 0;
-}
-
-void TheThingsNetwork::getInfo(String message, int infoType) {
   int i;
-  if (message == "") {
-    debugPrint("Information message empty");
-    return ;
+  String message = readValue(F("radio get sf"));
+  for (i = 2; message[i] && i <= 3; i++) {
+    this->info.sf = (this->info.sf + message[i] - 48) * 10;
   }
```

Comment:
What if we're calling this function twice; it's adding to the existing `sf`. Should the `sf` be reset before the loop?